lib/utils.ts

```diff
@@ -425,6 +425,41 @@ export const generateGravatarHash = (email: string | null): string => {
   return hash;
 };
 
+export async function generateEncrpytedPassword(
+  password: string,
+): Promise<string> {
+  if (!password) return "";
+  const encryptedKey: string = crypto
+    .createHash("sha256")
+    .update(String(process.env.NEXT_PRIVATE_DOCUMENT_PASSWORD_KEY))
+    .digest("base64")
+    .substring(0, 32);
+  const IV: Buffer = crypto.randomBytes(16);
+  const cipher = crypto.createCipheriv("aes-256-ctr", encryptedKey, IV);
+  let encryptedText: string = cipher.update(password, "utf8", "hex");
+  encryptedText += cipher.final("hex");
+  return IV.toString("hex") + ":" + encryptedText;
+}
+
+export function decryptEncrpytedPassword(password: string): string {
+  if (!password) return "";
+  const encryptedKey: string = crypto
+    .createHash("sha256")
+    .update(String(process.env.NEXT_PRIVATE_DOCUMENT_PASSWORD_KEY))
+    .digest("base64")
+    .substring(0, 32);
+  const textParts: string[] = password.split(":");
+  if (!textParts || textParts.length !== 2) {
+    return password;
```

Comment:
I changed it to return the hashed password. So it doesn't change for current users